ChatRPG/Services/GameInputHandler.cs

```diff
@@ -60,81 +66,118 @@ private void OnChatCompletionChunkReceived(bool isStreamingDone, string? chunk =
+                if (opponent == null)
+                {
+                    _logger.LogError("Opponent is unknown!");
+                    return _systemPrompts[SystemPromptType.DoAction];
+                }
+                SystemPromptType systemPromptType = DetermineCombatOutcome();
+                (int playerDmg, int opponentDmg) = ComputeCombatDamage(systemPromptType, opponent.Type);
+                string combatMessageContent = "";
+                if (playerDmg != 0)
+                {
+                    if (opponent.AdjustHealth(-playerDmg))
+                    {
+                        combatMessageContent +=
+                            $" With no health points remaining, {opponent.Name} dies and can no longer participate in the narrative.";
+                    }
+                    combatMessageContent += $"The player hits with their attack, dealing {playerDmg} damage. The opponent has {opponent.CurrentHealth} health remaining.";
+                    _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", campaign.Player.Name, opponent.Name, playerDmg, opponent.CurrentHealth, opponent.MaxHealth);
+                }
+                else
+                {
+                    combatMessageContent += $"The player misses with their attack, dealing no damage. The opponent has {opponent.CurrentHealth} health remaining.";
                 }
-                messageContent += $"The player hits with their attack, dealing {playerDmg} damage. The opponent has {opponent.CurrentHealth} health remaining.";
-                _logger.LogInformation("Combat: {Name} hits {Name} for {x} damage. Health: {CurrentHealth}/{MaxHealth}", campaign.Player.Name, opponent.Name, playerDmg, opponent.CurrentHealth, opponent.MaxHealth);
-            }
-            else
-            {
-                messageContent += $"The player misses with their attack, dealing no damage. The opponent has {opponent.CurrentHealth} health remaining.";
-            }
 
-            if (opponentDmg != 0)
-            {
-                messageContent += $"The opponent will hit with their next attack, dealing {opponentDmg} damage. The player has {campaign.Player.CurrentHealth} health remaining.";
-                if (campaign.Player.AdjustHealth(-opponentDmg))
+                if (opponentDmg != 0)
                 {
-                    await HandlePlayerDeath(campaign.Player, conversation);
+                    combatMessageContent += $"The opponent will hit with their next attack, dealing {opponentDmg} damage. The player has {(campaign.Player.CurrentHealth - opponentDmg < 0 ? 0 : campaign.Player.CurrentHealth - opponentDmg)} health remaining.";
+                    if (campaign.Player.AdjustHealth(-opponentDmg))
+                    {
+                        combatMessageContent += "The player has died and their adventure ends.";
+                    }
```

Comment:
Why not:
```suggestion
                    bool playerDied = campaign.Player.AdjustHealth(-opponentDmg);
                    combatMessageContent += $"The opponent will hit with their next attack, dealing {opponentDmg} damage. The player has { campaign.Player.CurrentHealth } health remaining.";
                    if (playerDied)
                    {
                        combatMessageContent += "The player has died and their adventure ends.";
                    }
```
Alternativt kan du også bare bruge `Math.Max(0, campaign.Player.CurrentHealth - opponentDmg)` for at gøre det kortere.

Reply:
donezo